src/components/modal/index.d.ts

```diff
@@ -0,0 +1,35 @@
+import React from "react"
+
+export type ModalContentProps = {
+  height: number | string
+  width: number | string
+  testId: string
+}
+
+export type ModalHeaderProps = {
+  testId: string
+}
+
+export type ModalBodyProps = {
+  testId: string
+}
+
+export type ModalFooterProps = {
+  testId: string
```

Comment:
Just a thought here, couldn't we have `testId` (as optional) as a generic type and just import it and extend types where needed?

Reply:
Agree with the optional, but what do you mean as generic type?

Reply:
I believe to make have a common interface/type and enhance the component specific interfaces/types?

Reply:
We can create a  `type PropsWithTest<P> = P & {testId?:string}` this will be simillar on how react is handling props when using `React.FC`